### ISSUE TITLE
complete the JavaScript cursor tests for `r.expr`

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#5473]
The tests for `r.expr` in `cursor.mocha` are incomplete, at least as compared to those for `r.table` and `r.range`. These should be filled in to be as comparable as possible. I don't want to hold up other tests for writing these, so this is reminder for myself to get back to them.

**Comment by danielmewes** [comment]
Yeah I think https://github.com/rethinkdb/rethinkdb/issues/5003 for example is a lot more important than this.

It actually looks like the driver is sharing a lot of code between a "real" cursor and an array result, so covering both isn't that immediately critical (despite indications that some things do *not* behave correctly for `r.expr([...])` https://github.com/rethinkdb/rethinkdb/issues/5469, but even those are not critical).
